cli/azd/pkg/environment/remote/remote.go

```diff
@@ -0,0 +1,224 @@
+	accountManager account.Manager
+	azCli          azcli.AzCli
+}
+
+func NewDefaultEnvironment(
+	console input.Console,
+	accountManager account.Manager,
+	azCli azcli.AzCli,
+) Environment {
+	return &DefaultEnvironment{
+		console:        console,
+		accountManager: accountManager,
+		azCli:          azCli,
+	}
+}
+
+func (e *DefaultEnvironment) PromptSubscription(ctx context.Context, msg string) (subscriptionId string, err error) {
+	subscriptionOptions, defaultSubscription, err := e.getSubscriptionOptions(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	if len(subscriptionOptions) == 0 {
+		return "", fmt.Errorf(heredoc.Doc(
+			`no subscriptions found.
+			Ensure you have a subscription by visiting https://portal.azure.com and search for Subscriptions in the search bar.
+			Once you have a subscription, run 'azd auth login' again to reload subscriptions.`))
+	}
+
+	for subscriptionId == "" {
+		subscriptionSelectionIndex, err := e.console.Select(ctx, input.ConsoleOptions{
+			Message:      msg,
+			Options:      subscriptionOptions,
+			DefaultValue: defaultSubscription,
+		})
+
+		if err != nil {
+			return "", fmt.Errorf("reading subscription id: %w", err)
+		}
+
+		subscriptionSelection := subscriptionOptions[subscriptionSelectionIndex]
+		subscriptionId = subscriptionSelection[len(subscriptionSelection)-
+			len("(00000000-0000-0000-0000-000000000000)")+1 : len(subscriptionSelection)-1]
+	}
+
+	if !e.accountManager.HasDefaultSubscription() {
+		if _, err := e.accountManager.SetDefaultSubscription(ctx, subscriptionId); err != nil {
+			log.Printf("failed setting default subscription. %s\n", err.Error())
+		}
+	}
+
+	return subscriptionId, nil
+}
+
+func (e *DefaultEnvironment) PromptLocation(
+	ctx context.Context,
+	subId string,
+	msg string,
+	filter LocationFilterPredicate,
+) (string, error) {
+	loc, err := promptLocationWithFilter(ctx, subId, msg, "", e.console, e.accountManager, filter)
+	if err != nil {
+		return "", err
+	}
+
+	if !e.accountManager.HasDefaultLocation() {
+		if _, err := e.accountManager.SetDefaultLocation(ctx, subId, loc); err != nil {
+			log.Printf("failed setting default location. %s\n", err.Error())
+		}
+	}
+
+	return loc, nil
+}
+
+func (e *DefaultEnvironment) PromptResourceGroup(ctx context.Context, subId string) (string, error) {
+	// Get current resource groups
+	groups, err := e.azCli.ListResourceGroup(ctx, subId, nil)
+	if err != nil {
+		return "", fmt.Errorf("listing resource groups: %w", err)
+	}
+
+	slices.SortFunc(groups, func(a, b azcli.AzCliResource) int {
+		return strings.Compare(a.Name, b.Name)
+	})
+
+	choices := make([]string, len(groups)+1)
+	for idx, group := range groups {
+		choices[idx] = fmt.Sprintf("%d. %s", idx, group.Name)
+	}
+
+	choice, err := e.console.Select(ctx, input.ConsoleOptions{
+		Message: "Pick a resource group to use:",
+		Options: choices,
+	})
+	if err != nil {
+		return "", fmt.Errorf("selecting resource group: %w", err)
+	}
+
+	return groups[choice].Name, nil
+}
```

Comment:
We already have this prompting logic in the provision providers.  We will want to keep things as DRY as possible and reuse these components.

Reply:
The one we already have relies on the environment package which creates circular dependency if I attempt to use it.

The solution to it is easy but I didn't want to change the current code as I don't know if it will affect other things.
The Prompter Package doesn't need to rely on the environment in getting the subscription or the location as it's only getting constants from it and that's for the location only.

Location use from the environment package:
![image](https://github.com/Azure/azure-dev/assets/64026625/c0baf5b6-d187-4dcf-baab-1ecff6bb9531)

Only the prompt resource group function needs the environment package as it heavily relies on it.

And I don't think the current implementation for prompt resource group will work as it gets everything from the environment which doesn't exist for this current scenario so, even if we found a way around the circular dependency we will still need to implement a new function for the prompt resource group or save the environment to the local env and check if found leave it if not delete it (which I don't think a very good approach)